packages/foundations/scss/icon/icons.scss

```
@@ -1,33 +1,91 @@
-@use "icons.helpers" as *;
-@use "icons.placeholder" as *;
-@use "../variables.global" as *;
-@forward "icons.attributes-mappings";
-@forward "icons.custom-properties";
-@forward "icons.font-faces";
+@use "icons.helpers" as icon-helper;
+@use "../variables.global" as var;
+@use "../default.assets-paths" as *;
+
+@each $variant in icon-helper.$default-icon-variants {
+	@font-face {
+		font-display: block;
+		font-family: "db-ux-#{$variant}";
+		font-style: normal;
+		font-weight: normal;
+		src: url("#{$icons-path}functional/fonts/#{$variant}/db-ux.woff2")
+			format("woff2");
+	}
+
+	@each $size in (16, 20, 24, 32, 48, 64) {
+		@font-face {
+			font-display: block;
+			font-family: "db-ux-#{$variant}";
+			font-style: normal;
+			font-weight: $size;
+			src: url("#{$icons-path}functional/fonts/#{$variant}_#{$size}/db-ux.woff2")
+				format("woff2");
+		}
+
+		[data-icon-weight="#{$size}"],
+		[data-icon-weight-before="#{$size}"] {
+			&::before {
+				--icon-font-weight: #{$size};
+			}
+		}
+
+		[data-icon-weight-after="#{$size}"] {
+			&::after {
+				--icon-font-weight: #{$size};
+			}
+		}
```

```suggestion
		[data-icon-size="#{$size}"],
		[data-icon-size-before="#{$size}"] {
			&::before {
				--icon-font-size: #{$size};
			}
		}

		[data-icon-size-after="#{$size}"] {
			&::after {
				--icon-font-size: #{$size};
			}
		}
```

I'm struggling with calling this `weight`, as we mainly use this funky solution under the hood, but it might be confusing for others using our components, as it might be `size` even also in the docs.

---

But we have a `--icon-font-size` as well to control the real font-size...

I would keep weight because that's what we change here.